okrs_2019.md

```diff
@@ -0,0 +1,209 @@
+- transferring context on the project, its history and design
+- writing documentation
+- improving test coverage
+- maintaining the build
+- triaging issues
+- helping to resolve interpersonal conflicts
+- planning for future development
+- refactoring difficult code
+- paying down technical debt
+- making the project fun to work on
+
+### Initiatives
+- * [ ] Add code analysis to cred
+  - * [ ] Individual files are tracked by SourceCred
+  - *  [ ] Call graph analysis for JS code
+  - * [ ] Call graph analysis for Go code
+- * [ ] Add time-based cred analysis
+  - * [ ] Able to display cred for specific time ranges (e.g. Jan 2019)
+  - * [ ] Able to apply overall weighting function based on when a contribution was made
+- * [ ] Add support for heuristics to SourceCred
+- * [ ] Add heuristics or plugins for detecting several forms of project stewardship
+- * [ ] Come up with robust metrics for measuring attribution quality
+- * [ ] Add 'cred decomposition'
+
+  (e.g. who has cred for a particular file, or for documentation work in general)
+
+### Metrics
+
+Coming up with metrics for attribution quality is challenging, since there
+isn't a readily available ground truth to compare against. Right now, the best
+metric we have is asking the contributors to a project to rate the quality of
+our attribution. The "cred rating" in the table below is the average rating
+that contributors to a project give its cred attribution, measured on a
+10-point scale.
+
+We also would like to include explicit heuristics for a number of the 'Project
+Stewardship' tasks described above.
+
+| Metric                                  | Goal | Q4 | Q1 | Q2 | Q3 | Q4 |
+| --------------------------------------- | ---- | -- | -- | -- | -- | -- |
+| SourceCred's cred rating                | 8/10 | -  |    |    |    |    |
+| IPFS's cred rating                      | 7/10 | -  |    |    |    |    |
+| # of measured Project Stewardship Tasks | 3    | 0  |    |    |    |    |
+| # of initiatives completed              | 6    | 0  |    |    |    |    |
+
+
+## SourceCred Community Growth
+
+We envision SourceCred being a thriving open-source community which is
+passionate about recognizing and rewarding open-source contributors, about the
+art of software engineering itself, and, of course, about building SourceCred.
+
+Right now we have a pretty good start in that the founders are passionate about
+these things ;) but we hope to grow the engaged community of contributors and
+users.
+
+On the initiative side, we want to improve the documentation around all parts of
+SourceCred so that the project is easier to get up to speed on. We also intend
+to produce content explaining SourceCred, like blog posts and video talks.
+
+### Initiatives
+- * [ ] Improve SourceCred's documentation
+  - * [ ] Document the overall vision, mission, and philosophy
+  - * [ ] Document the overall technical architecture
+  - * [ ] Document the attribution algorithm (e.g. by writing a whitepaper)
```

Comment:
> Document the attribution algorithm (e.g. by writing a whitepaper)

Suggest deprioritizing unless and until we have something interesting.
What we have now does not merit a white paper; it merits a paragraph in
our general documentation. There’s nothing special about it.

Reply:
Sure thing. I amended the "improve cred attribution" to more clearly express that many of the initiatives are in fact research problems, and we should start by clearly documenting those problems. That is actually what I was getting at here.